### PR TITLE
Fix regenerating recovery codes webauthn scope

### LINKED ISSUE #40528: Recreating recovery codes through the account settings page doesn't work
Expected behavior:
User should be able to recreate the recovery codes through the settings page.

Current behavior:
A following error appears:

> MFA response validation failed: required scope "CHALLENGE_SCOPE_MANAGE_DEVICES" is not satisfied by the given webauthn session with scope "CHALLENGE_SCOPE_ACCOUNT_RECOVERY"

Creating recovery codes while setting up accounts or resetting authentication works correctly.

Bug details:
Teleport version: v15, master

### PR DESCRIPTION
**Pull request by Joerger**
Changelog: Fix regenerating cloud account recovery codes

The `CreatePrivilegeToken` endpoint always expects the `MANAGE_DEVICES` webauthn scope. Since regenerating recovery codes uses a privileged token instead of using a webauthn response directly, using the `ACCOUNT_RECOVERY` scope for this flow would result in an error:

image

Note: `ACCOUNT_RECOVERY` is exclusively expected by the `VerifyAccountRecovery` endpoint. It should be used for other one-shot recovery endpoints as well.

Fixes https://github.com/gravitational/teleport/issues/40528

TODO: follow up on `TODO` once https://github.com/gravitational/teleport.e/pull/3980 is merged as well.

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]
This has regressed more than once, so it would be great to figure out a way to test it.

**Comment by Joerger** [comment]
> This has regressed more than once, so it would be great to figure out a way to test it.

I've refactored the privilege-token-based re-authentication flow to ensure the `MANAGE_DEVICES` scope expected by `rpc CreatePrivilegeToken` is always used.

Other than that, we'd need to add an e2e test that checks that the scope provided on the front end matches the scope expected on the backend, which I'm not sure is necessary.

**Review by codingllama — COMMENTED** [review]
Seems reasonable to me, but I think a frontend person should take a look.

File: web/packages/teleport/src/components/ReAuthenticate/useReAuthenticate.ts [hunk 1]
```diff
@@ -53,12 +53,12 @@ export default function useReAuthenticate(props: Props) {
...
   }
 
-  function submitWithWebauthn(scope: MfaChallengeScope) {
+  function submitWithWebauthn() {
     setAttempt({ status: 'processing' });
 
     if ('onMfaResponse' in props) {
       auth
-        .getWebauthnResponse(scope)
+        .getWebauthnResponse(props.challengeScope)
```
Comment (bl-nero) [inline web/packages/teleport/src/components/ReAuthenticate/useReAuthenticate.ts:61]: Could we document the `challengeScope` field as irrelevant if we are going this way? (Ideally, the presence of an event handler should not even affect the logic here, but for the sake of making minimal changes, we can omit this.)

**Review by bl-nero — APPROVED** [review]

File: web/packages/teleport/src/components/ReAuthenticate/useReAuthenticate.ts (hunk 1 above)
Comment (Joerger) [reply web/packages/teleport/src/components/ReAuthenticate/useReAuthenticate.ts:61]: The challengeScope is relevant here for `getWebauthnResponse`, it's just not relevant for `createPrivilegeTokenWithWebauthn`.

I agree that using the event handler's presence as a logic switch is odd, but the way it actually works is that `props` can only be one of two different mutually exclusive types:

```ts
// MfaResponseProps defines a function
// that accepts a MFA response. No
// authentication has been done at this point.
type MfaResponseProps = BaseProps & {
  onMfaResponse(res: MfaAuthnResponse): void;
  /**
   * The MFA challenge scope of the action to perform, as defined in webauthn.proto.
   */
  challengeScope: MfaChallengeScope;
  onAuthenticated?: never;
};

// DefaultProps defines a function that
// accepts a privilegeTokenId that is only
// obtained after MFA response has been
// validated.
type DefaultProps = BaseProps & {
  onAuthenticated(privilegeTokenId: string): void;
  onMfaResponse?: never;
  // TODO(Joerger): change type to 'never' once it is no longer expected in /e
  challengeScope?: MfaChallengeScope;
};
```

I'm not sure what additional info to document, so feel free to leave any specific suggestions, or resolve this comment if you're ok with merging as is.

**Pull request merged**